packages/client/screens/account/AccountNameScreen.tsx

```diff
@@ -0,0 +1,115 @@
+        try {
+            const user = await usersGateway.update(values, token);
+            setUser(user);
+
+            toast.show({
+                placement: "top",
+                duration: 2000,
+                render: () => {
+                    return (
+                        <Box style={styles.toastBox}>
+                            <Text size={"footnote1"} fontWeight={"semibold"} color={"white"} style={styles.toastText}>
+                                Name updated succesfully
+                            </Text>
+                        </Box>
+                    );
+                },
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                formikHelpers.setFieldError("firstName", error.message);
+                formikHelpers.setFieldError("lastName", error.message);
+            }
+        }
+    }
+
+    return (
+        <View style={styles.view}>
+            <Formik initialValues={intitialValues} validationSchema={updateUserSchema} onSubmit={handleUpdate}>
+                {({ values, errors, touched, handleChange, submitForm }) => (
+                    <VStack space={4} marginTop={5}>
+                        <FormControl isInvalid={!!(errors.firstName && touched.firstName)}>
+                            <Input
+                                value={values.firstName}
+                                onChangeText={handleChange("firstName")}
+                                placeholder="First name"
+                                testID="firstName"
+                            />
+                            <FormControl.ErrorMessage>{touched.firstName && errors.firstName}</FormControl.ErrorMessage>
+                        </FormControl>
+                        <FormControl isInvalid={!!(errors.lastName && touched.lastName)}>
+                            <Input
+                                value={values.lastName}
+                                onChangeText={handleChange("lastName")}
+                                placeholder="Last name"
+                                testID="lastName"
+                            />
+                            <FormControl.ErrorMessage>{touched.lastName && errors.lastName}</FormControl.ErrorMessage>
+                            <Button
+                                disabled={
+                                    intitialValues.firstName === values.firstName &&
+                                    intitialValues.lastName === values.lastName
+                                }
+                                style={
+                                    intitialValues.firstName === values.firstName &&
+                                    intitialValues.lastName === values.lastName
+                                        ? styles.ButtonDisabled
+                                        : styles.Button
+                                }
```

Comment:
You aren't using the right props, this is how it should be. The disabled styles are already defined

```suggestion
                                isDisabled={
                                    intitialValues.firstName === values.firstName &&
                                    intitialValues.lastName === values.lastName
                                }
                                style={styles.Button}
```